design/FY2019/10x EliminateIteration.md

```diff
@@ -0,0 +1,614 @@
+
+Figure 4 shows sensible heating capacity vs. cycle or speed ratio at 4 different speeds using constant volume fan and cycling coil. It is clear that the linear relationships are shown with all speeds.   
+
+The following table lists linear regression results at 4 different speeds. It contains values of slope, intercept and r2. The values of r2 at all speeds are 1.
+
+|           | Speed 1  | Speed 2  | Speed 3  | Speed 4  |
+|-----------|----------|----------|----------|----------|
+| Slope     | 2909.972 | 3027.975 | 1874.699 | 2342.797 |
+| Intercept | -1408.76 | 1501.222 | 4529.198 | 6403.896 |
+| r2        | 1        | 1        | 1        | 1        |
+    
+![Caption = "Figure 5 ..."](Fig5withCycFanCycCoilHeating.png)
+**Figure 5. Sensible heating capacity vs cycle or speed ratio at 4 different speeds with fan operation mode of cycling fan and cycling coil**
+
+Figure 5 shows sensible heating capacity vs. cycle or speed ratio at 4 different speeds using cycling fan and cycling coil. It is clear that the linear relationships are shown with all speeds.   
+
+The following table lists linear regression results at 4 different speeds. It contains values of slope, intercept and r2. The values of r2 at all speeds are 1 or very close to 1.
+|           | Speed 1  | Speed 2  | Speed 3  | Speed 4  |
+|-----------|----------|----------|----------|----------|
+| Slope     | 1349.446 | 2876.506 | 1723.328 | 2153.583 |
+| Intercept | 4.812044 | 1349.883 | 4226.387 | 5949.715 |
+| r2        | 0.999971 | 1        | 1        | 1        |
+
+The multispeed heat pump has a supplemental heater. Since the heater does not require iteration, the existing code can be kept.
+
+**Theoretically, the proposed approach provides more accurate or exact solution. The iteration approach provides approximated solution within convergence limit. In addition, the proposed approach will not have any divergence issues with very small load, while the iteration approach may get divergence.** 
+
+**Furthermore, the proposed approach will eliminate a section of code to handle iteration about 10 - 20%.**
+
+## Approach ##
+
+We propose that the iterations may be eliminated by calculating part load ratios and speed ratios directly. Specifically, for a single speed coil, the system load has to be met by a coil:
+
+Load = PLR * ( SysCap at full load – SysCap at PLR=0 )
+
+The same calculation procedure is used at the lowest speed for a multiple speed. For a multispeed coil, the system load has to be met by a coil:
+
+Load = SysCap1 at full load * SR + SysCap2 at full load * (1 – SR)
+
+The above two equations show a linear relationship between SysCap and PLR or SR. Therefore, PLR and SR can be calculated directly as long as system full capacities are given. If a proposed solution strategy is applied, it should speed up EnergyPlus calculations dramatically.
+
```

Comment:
The coil PLF curve needs to be used by the fan for this to work. Currently it's passed to the fan via global FanOnOffPartLoadFrac, And if the fan is blow thru, it would take 2 iterations to get the answer since the coil sets this global, unless the fan knew the PLR before the coil was simulated.

Reply:
@rraustad After calculating PLR, I call the Calc* function again to ensure all node get proper properties. Here is a calling sequence

                            PartLoadFrac = (QZnReq - TempOutput0) / (MSHeatPump(MSHeatPumpNum).FullOutput(i) - TempOutput0);
                            CalcMSHeatPump(MSHeatPumpNum,
                                FirstHVACIteration,
                                CompOp,
                                i,
                                0.0,
                                PartLoadFrac,
                                TempOutput,
                                QZnReq,
                                OnOffAirFlowRatio,
                                SupHeaterLoad);

I tested two fan placements and got reasonable results. I will check it again to ensure fan operation will be correct.